src/System.Buffers.Primitives/System/Buffers/ReaderWriter/BufferReader_search.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Sequences;
+
+namespace System.Buffers
+{
+    public ref partial struct BufferReader<TSequence> where TSequence : ISequence<ReadOnlyMemory<byte>>
+    {
+        public bool TryReadUntill(out ReadOnlyBuffer bytes, byte delimiter)
+        {
+            var range = new PositionRange(CurrentPosition, AdvanceToDelimiter(delimiter).GetValueOrDefault());
+            if (range.Start == default || range.End == default)
+            {
+                bytes = default;
+                return false;
+            }
+
+            bytes = new ReadOnlyBuffer(range.Start, range.End);
```

Comment:
This is not correct. You cannot just create `ROB` from random positions.

Reply:
Why not? Can we fix it in ROB?

Reply:
Position is transparent, the implementor of ISequence can put random things there. `ROB ` cannot support them all.